security/security.c

```diff
@@ -696,27 +777,29 @@ static void __init lsm_early_task(struct task_struct *task)
  * call_int_hook:
  *	This is a hook that returns a value.
  */
+#define CALL_STATIC_SLOT_VOID(NUM, HOOK, ...)				\
+	static_call_cond(STATIC_SLOT(HOOK, NUM))(__VA_ARGS__);
 
-#define call_void_hook(FUNC, ...)				\
-	do {							\
-		struct security_hook_list *P;			\
-								\
-		hlist_for_each_entry(P, &security_hook_heads.FUNC, list) \
-			P->hook.FUNC(__VA_ARGS__);		\
+#define call_void_hook(FUNC, ...)					\
+	do {								\
```

Comment:
Can't we have the `switch` for void hooks too?

Reply:
Sounds good